tests/test_virtualscreening.py

```diff
@@ -152,3 +163,83 @@ def test_vs_similarity():
+    for pdbbind_v in pdbbind_versions:
+        version_dir = os.path.join(data_dir, 'v%s' % pdbbind_v)
+        if not os.path.isdir(version_dir):
+            os.symlink(pdbbind_dir, version_dir)
+
+    filenames = []
+    # train mocked SFs
+    for model in [nnscore(n_jobs=1)] + [rfscore(version=v, n_jobs=1)
+                                        for v in [1, 2, 3]]:
+            model.gen_training_data(data_dir, pdbbind_versions=pdbbind_versions,
+                                    home_dir=home_dir)
+            filenames.append(model.train(home_dir=home_dir))
+    vs = virtualscreening(n_cpu=-1)
+    vs.load_ligands('sdf', xiap_actives_docked)
+    # error if no protein is fed
+    assert_raises(ValueError, vs.score, 'nnscore')
+    # bad sf name
+    assert_raises(ValueError, vs.score, 'bad_sf', protein=protein)
+    vs.score('nnscore', protein=xiap_protein)
+    vs.score('nnscore_pdbbind2016', protein=protein)
+    vs.score('rfscore_v1', protein=protein)
+    vs.score('rfscore_v1_pdbbind2016', protein=protein)
+    vs.score('rfscore_v2', protein=protein)
+    vs.score('rfscore_v3', protein=protein)
+    # use pickle directly
+    vs.score(filenames[0], protein=protein)
+    # pass SF object directly
+    vs.score(scorer.load(filenames[0]), protein=protein)
+    # pass wrong object (sum is not an instance of scorer)
+    assert_raises(ValueError, vs.score, sum, protein=protein)
+
+    mols = list(vs.fetch())
+
+    assert_equal(len(mols), 100)
+    mol_data = mols[0].data
+    assert_in('nnscore', mol_data)
+    assert_in('rfscore_v1', mol_data)
+    assert_in('rfscore_v2', mol_data)
+    assert_in('rfscore_v3', mol_data)
+
+    vs = virtualscreening(n_cpu=1)
+    vs.load_ligands('sdf', xiap_actives_docked)
+    vs.score('nnscore', protein=protein)
+    vs.score('rfscore_v1', protein=protein)
+    vs.score('rfscore_v2', protein=protein)
+    vs.score('rfscore_v3', protein=protein)
+    with NamedTemporaryFile('w', suffix='.sdf') as molfile:
+        with NamedTemporaryFile('w', suffix='.csv') as csvfile:
+            vs.write('sdf', molfile.name, csv_filename=csvfile.name)
+            data = pd.read_csv(csvfile.name)
+            assert_in('nnscore', data.columns)
+            assert_in('rfscore_v1', data.columns)
+            assert_in('rfscore_v2', data.columns)
+            assert_in('rfscore_v3', data.columns)
+
+            mols = list(oddt.toolkit.readfile('sdf', molfile.name))
+            assert_equal(len(mols), 100)
+
+            vs.write_csv(csvfile.name, fields=['nnscore', 'rfscore_v1',
+                                               'rfscore_v2', 'rfscore_v3'])
+            data = pd.read_csv(csvfile.name)
+            assert_equal(len(data.columns), 4)
+            assert_in('nnscore', data.columns)
+            assert_in('rfscore_v1', data.columns)
+            assert_in('rfscore_v2', data.columns)
+            assert_in('rfscore_v3', data.columns)
+
+    for f in filenames:
+        os.unlink(f)
```

Comment:
What about symlinks created for different PDBbind versions?

Reply:
Good point